From 372c6d3d0119e1c0b38f9eed0d7e9b4e45c355a3 Mon Sep 17 00:00:00 2001
From: Daniel Lemos <xspager@gmail.com>
Date: Tue, 15 Mar 2022 13:48:34 -0300
Subject: [PATCH 1/2] Setup the serenity platform

Setup serenity as a platform and configure `defaults.variables.UNZIP` to
work around our `unzip` command not supporting the flag `-n`
---
 src/luarocks/core/cfg.lua | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/luarocks/core/cfg.lua b/src/luarocks/core/cfg.lua
index 926d479..7c2914b 100644
--- a/src/luarocks/core/cfg.lua
+++ b/src/luarocks/core/cfg.lua
@@ -42,6 +42,7 @@ local platform_order = {
    "cygwin",
    "msys",
    "haiku",
+   "serenity",
    -- Windows
    "windows",
    "win32",
@@ -152,6 +153,7 @@ local platform_sets = {
    netbsd = { unix = true, bsd = true, netbsd = true },
    haiku = { unix = true, haiku = true },
    linux = { unix = true, linux = true },
+   serenity = { unix = true, serenity = true },
    mingw = { windows = true, win32 = true, mingw32 = true, mingw = true },
    msys = { unix = true, cygwin = true, msys = true },
    msys2_mingw_w64 = { windows = true, win32 = true, mingw32 = true, mingw = true, msys = true, msys2_mingw_w64 = true },
@@ -497,6 +499,12 @@ local function make_defaults(lua_version, target_cpu, platforms, home)
       defaults.variables.MAKE = "gmake"
    end
 
+   if platforms.serenity then
+      defaults.arch = "serenity"..target_cpu
+      defaults.variables.UNZIP = "unzip"
+      defaults.web_browser = "br"
+   end
+
    -- Expose some more values detected by LuaRocks for use by rockspec authors.
    defaults.variables.LIB_EXTENSION = defaults.lib_extension
    defaults.variables.OBJ_EXTENSION = defaults.obj_extension
-- 
2.36.1

